### PR TITLE
Add newtypes for BlockHeight/EpochLength/etc

### LINKED ISSUE #152: [100 SigmaUSD] Newtypes for epoch id/counter, datapoint, height
### Motivation
To not confuse them in function parameters with each other and with other integer values.

### Implementation details
Use the newtypes in the relevant structs, function parameters, etc.
Underneath, epoch id/counter should be u32, datapoint - i64, height - u32.

### PR DESCRIPTION
**Pull request by SethDusek**
Closes #152

### CODE REVIEW

File: core/src/cli_commands/vote_update_pool.rs [hunk 1]
```diff
@@ -323,7 +325,7 @@ mod tests {
             new_pool_box_address_hash,
             new_reward_token_id,
             100_000,
-            height - 3,
+            BlockHeight(height.0 - 3), // TODO: should BlockHeight implement Sub for arbitrary units?
```
Comment (SethDusek) [inline core/src/cli_commands/vote_update_pool.rs:328]: What do you think?

Comment (greenhat) [reply core/src/cli_commands/vote_update_pool.rs:328]: Sounds nice! If it does not take too much effort. :) I mean, bounds checks are helpful, but the main reason for this newtype is API.

**Comment by SethDusek** [comment]
Hmm, other than the coveralls pipeline failing (#225), this should be ready to review

File: core/src/oracle_types.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+use serde::Deserialize;
+use serde::Serialize;
+
+#[derive(PartialEq, PartialOrd, Eq, Ord, Debug, Serialize, Deserialize, Copy, Clone)]
+#[serde(transparent)]
+pub struct BlockHeight(pub u32);
+
+impl std::ops::Sub<EpochLength> for BlockHeight {
```
Comment (greenhat) [inline core/src/oracle_types.rs:8]: Nice!

**Review by greenhat — APPROVED** [review]
Looking great! Thank you!

I think generic ergo types like `BlockHeight` will eventually make their way into the ergo-lib.

**Pull request merged**